src/besselj.jl

```diff
@@ -150,13 +150,43 @@ function besselj1(x::Float32)
         return p * s
     end
 end
-"""
-    besselj(nu, x::T) where T <: Union{Float32, Float64}
 
-Bessel function of the first kind of order nu, ``J_{nu}(x)``.
-Nu must be real.
-"""
-function _besselj(nu, x)
+function besselj(nu::Real, x::T) where T
+    isinteger(nu) && return besselj(Int(nu), x)
+    abs_nu = abs(nu)
+    abs_x = abs(x)
+
+    Jnu = besselj_positive_args(abs_nu, abs_x)
+    if nu >= zero(T)
+        return x >= zero(T) ? Jnu : Jnu * cispi(abs_nu)
+    else
+        Ynu = bessely_positive_args(abs_nu, abs_x)
+        spi, cpi = sincospi(abs_nu)
+        out = Jnu * cpi - Ynu * spi
+        return x >= zero(T) ? out : out * cispi(nu)
+    end
+end
+
+function besselj(nu::Int, x::T) where T
+    abs_nu = abs(nu)
+    abs_x = abs(x)
+    sg = iseven(Int(abs_nu)) ? 1 : -1
```

Comment:
`Int` is unnecessary here.

Reply:
I think on v1.6 `iseven` only had methods for Int? It failed tests originally without it on that version.

Reply:
right, but then you split it up, so `nu` is always an `Integer` so this is valid again.

Reply:
Haha yep..... 🤷‍♂️